lang/syn/src/idl/defined.rs

```diff
@@ -424,6 +430,9 @@ pub fn gen_idl_type(
         syn::Type::Path(path) if the_only_segment_is(path, "Pubkey") => {
             Ok((quote! { #idl::IdlType::Pubkey }, vec![]))
         }
+        syn::Type::Path(path) if the_only_segment_is(path, "PhantomData") => {
+            Ok((TokenStream::new(), vec![]))
+        }
```

Comment:
Nit: `match` arms more or less follow the order of [`IdlType`](https://github.com/coral-xyz/anchor/blob/6df05aa5b5d9971041e3d25a246d935252b6e576/idl/spec/src/lib.rs#L282), so it would be great if you could move this logic either inside the `Defined` impl:

 https://github.com/coral-xyz/anchor/blob/6df05aa5b5d9971041e3d25a246d935252b6e576/lang/syn/src/idl/defined.rs#L484-L485
 
 or just above it.